notebooks/customizing_potentials_cookbook.ipynb

```diff
@@ -0,0 +1,1780 @@
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "3yVG0w4oBV1q",
+        "colab_type": "text"
+      },
+      "source": [
+        "plot $V(r)$"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "TgIbQa_D1kIM",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "drs = np.arange(0,3,0.01)\n",
+        "U = harmonic_morse(drs)\n",
+        "plt.plot(drs,U)\n",
+        "format_plot(r'$r$', r'$V(r)$')\n",
+        "finalize_plot()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "C1lMC0LeCSe9",
+        "colab_type": "text"
+      },
+      "source": [
+        "##Calculate the energy of a system of interacting particles"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "kyuIoCz3maxs",
+        "colab_type": "text"
+      },
+      "source": [
+        "We now want to calculate the energy of a system of $N$ spheres in $d$ dimensions, where each particle interacts with every other particle via our user-defined function $V(r)$. The total energy is\n",
+        "\\begin{equation}\n",
+        "E_\\text{total} = \\sum_{i<j}V(r_{ij}),\n",
+        "\\end{equation}\n",
+        "where $r_{ij}$ is the distance between particles $i$ and $j$. \n",
+        "\n",
+        "Our first task is to set up the system by specifying the $N$, $d$, and the size of the simulation box. We then use JAX's internal random number generator to pick positions for each particle. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "q77Cepyy4M14",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "N = 50\n",
+        "dimension = 2\n",
+        "box_size = 6.8\n",
+        "\n",
+        "key, split = random.split(key)\n",
+        "R = random.uniform(split, (N,dimension), minval=0.0, maxval=box_size, dtype=f64) \n",
+        "\n",
+        "plot_system(R,box_size)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "lB5ISBfB4txR",
+        "colab_type": "text"
+      },
+      "source": [
+        "At this point, we could manually loop over all particle pairs and calculate the energy, keeping track of boundary conditions, etc. Fortunately, JAX MD has machinery to automate this. \n",
+        "\n",
+        "First, we must define two functions, ```displacement``` and ```shift```, which contain all the information of the simulation box, boundary conditions, and underlying metric. ```displacement``` is used to calculate the vector displacement between particles, and ```shift``` is used to move particles. While we can define these manually, for many common cases JAX MD has them built in. We will use a simple square box with periodic boundary conditions. "
```

Comment:
Done. I took out all the error handling, etc. so that they are more easily understood, and I added a note encouraging people to use the built in functionality when possible.